Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -1290,14 +1319,15 @@ void MediaPlayerPrivateGStreamerBase::handleProtectionStructure(const GstStructu
 void MediaPlayerPrivateGStreamerBase::cdmInstanceAttached(const CDMInstance& instance)
 {
     ASSERT(isMainThread());
+    LockHolder lock(m_protectionMutex);
     GST_DEBUG("CDM instance %p set", &instance);
     m_cdmInstance = &instance;
-    dispatchLocalCDMInstance();
+    dispatchLocalCDMInstance(false);
 }
 
-void MediaPlayerPrivateGStreamerBase::dispatchLocalCDMInstance()
+void MediaPlayerPrivateGStreamerBase::dispatchLocalCDMInstance(bool shouldLock)
 {
-    ASSERT(isMainThread());
+    LockHolder lock(shouldLock ? &m_protectionMutex : nullptr);
```

Comment:
LockHolder as a constructor for pointers and references. Here I prefer to use the pointer one to be able to pass nullptr if we don't have to lock. Either that or locking and unlocking manually. I thought it this way it would be more clear and safer.